sky/utils/command_runner.py

```diff
@@ -198,7 +213,8 @@ def make_runner_list(
             port_list = [22] * len(ip_list)
         return [
             SSHCommandRunner(ip, ssh_user, ssh_private_key, ssh_control_name,
```

Comment:
On GKE clusters, This PR is currently using the external IP of nodes (`sky@34.16.78.81`):
```
2023-08-29 17:53:01,300	VVINFO command_runner.py:367 -- Full command is `[1mssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_8d2ba49c09/098f6bcd46/%C -o ControlPersist=10s -o ProxyCommand=ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -p 23100 -W %h:%p sky@34.16.78.81 -o ProxyCommand='/Users/romilb/.sky/port-forward-proxy-cmd.sh'  -o Port=22 -o ConnectTimeout=120s sky@10.72.0.17 bash --login -c -i 'true && source ~/.bashrc && export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (mkdir -p ~/.sky/.runtime_files)'[22m[26m`
building file list ... done
```
This will not work if the nodes are behind a firewall. It should instead of be connecting to `sky@127.0.0.1` since the the port-forward is running locally. 

You may need to update get_external_ip to accept an arg with the `KubernetesNetworkingMode`, which would return 127.0.0.1 if the mode is port-forward, else use the existing logic if the mode is nodeport.